Pod/Classes/PINRemoteImageBasicCache.m

```diff
@@ -0,0 +1,93 @@
+//
+//  PINRemoteImageBasicCache.m
+//  Pods
+//
+//  Created by Aleksei Shevchenko on 7/28/16.
+//
+//
+
+#import "PINRemoteImageBasicCache.h"
+
+@interface PINRemoteImageBasicCache()
+@property (nonatomic, strong) NSCache *cache;
+@end
+
+@implementation PINRemoteImageBasicCache
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        self.cache = [[NSCache alloc] init];
+    }
+    return self;
+}
+
+//******************************************************************************************************
+// Memory cache methods
+//******************************************************************************************************
+-(nullable id)objectCachedInMemoryForKey:(NSString *)key
+{
+    return [self.cache objectForKey:key];
+}
+
+-(void)cacheObjectInMemory:(id)object forKey:(NSString *)key withCost:(NSUInteger)cost
+{
+    [self.cache setObject:object forKey:key cost:cost];
+}
+
+//******************************************************************************************************
+// Disk cache methods
+//******************************************************************************************************
+-(nullable id)objectCachedOnDiskForKey:(NSString *)key
+{
+    return [self.cache objectForKey:key];
+}
+
+-(void)objectCachedOnDiskForKey:(NSString *)key block:(PINRemoteImageCachingObjectBlock)block
+{
+    __weak typeof(self) welf = self;
+    dispatch_async(dispatch_get_global_queue(QOS_CLASS_DEFAULT, 0), ^{
```

Comment:
I believe `QOS_CLASS_DEFAULT` is only available on iOS8+? We should either add a custom macro if `QOS_CLASS_DEFAULT` is not defined or fallback to something that at least works with iOS7. If we add the macro we should satisfy the remaining QOS levels as well

```
 * A QOS class value:
 *  - QOS_CLASS_USER_INTERACTIVE
 *  - QOS_CLASS_USER_INITIATED
 *  - QOS_CLASS_DEFAULT
 *  - QOS_CLASS_UTILITY
 *  - QOS_CLASS_BACKGROUND
 *  - QOS_CLASS_UNSPECIFIED
```